awscli/customizations/eks/kubeconfig.py

```diff
@@ -205,33 +205,28 @@ def write_kubeconfig(self, config):
 class KubeconfigAppender(object):
     def insert_entry(self, config, key, entry):
         """
-        Insert entry into the array at content[key]
+        Insert entry into the entries list at content[key]
         Overwrite an existing entry if they share the same name
 
         :param config: The kubeconfig to insert an entry into
         :type config: Kubeconfig
         """
-        if key not in config.content:
-            config.content[key] = []
-        array = config.content[key]
-        if not isinstance(array, list):
-            raise KubeconfigError("Tried to insert into {0},"
-                                  "which is a {1} "
-                                  "not a {2}".format(key,
-                                                     type(array),
-                                                     list))
+        config.content[key] = config.content.get(key) or []
+        entries = config.content[key]
+        if not isinstance(entries, list):
+            raise KubeconfigError(f"Tried to insert into {key},"
+                                  f"which is a {type(entries)} "
+                                  f"not a {list}")
         found = False
-        for counter, existing_entry in enumerate(array):
-            if "name" in existing_entry and\
-               "name" in entry and\
-               existing_entry["name"] == entry["name"]:
-                array[counter] = entry
+        for i, existing_entry in enumerate(entries):
+            if "name" in existing_entry and "name" in entry \
```

Comment:
Since we are doing some general cleanup, I'd say this is a good time to make the top level pubic `insert_entry` method more readable by doing something like:
```python
    def insert_entry(self, config, key, entry):
        """
        Insert entry into the array at content[key]
        Overwrite an existing entry if they share the same name

        :param config: The kubeconfig to insert an entry into
        :type config: Kubeconfig
        """
        entries = self._get_existing_entries(key, config)
        existing_entry_index = self._index_of_entry_with_name(entry.get("name"), entries)

        if existing_entry_index >= 0:
            entries[existing_entry_index] = entry
        else:
            entries.append(entry)

        return config

    def _get_existing_entries(self, key, config):
        if key not in config.content:
            config.content[key] = []
        entries = config.content[key]
        if not isinstance(entries, list):
            raise KubeconfigError(
                f"Tried to insert into {key}, "
                f"which is a {type(entries)} "
                f"not a list"
            )
        return entries

    def _index_of_entry_with_name(self, name, entries):
        if name is None:
            return -1
        for i, entry in enumerate(entries):
            if entry.get("name") == name:
                return i
        return -1

```

Reply:
Thank you I'll tackle these changes. 

While we are at it, what do you think about the OOP style here? Frankly I really dislike it, this isn't Python. For example, all the methods in this `KubeconfigAppender` class should be static methods, which in turn should not even make up a *class*.

Reply:
Overall I used your code, but opted to not change the current logic for names with `None` values.